apps/submission/tests/io/test_pixel.py

```diff
@@ -291,6 +291,37 @@ def test_save(self):
         parser.save()
         self.assertEqual(Pixel.objects.count(), 1837)
 
+    def test_save_with_same_reference_omics_units(self):
+
+        # Create two OmicsUnits with the same identifier but for different 
+        # strains/types
+        OmicsUnitFactory(
+            reference=EntryFactory(identifier='CAGL0B00990g'),
+            strain=StrainFactory(),
+            type=OmicsUnitTypeFactory()
+        )
+        OmicsUnitFactory(
+            reference=EntryFactory(identifier='CAGL0B00990g'),
+            strain=self.strain,
+            type=self.omics_unit_type
+        )
+
+        parser = PixelSetParser(
+            Path(
+                'apps/submission/fixtures/dataset-0002/data.txt'
+            ),
+            description=self.description,
+            analysis=self.analysis,
+            omics_unit_type=self.omics_unit_type,
+            strain=self.strain,
+        )
+        parser.parse()
+        self._load_cgd_entries()
+
+        self.assertEqual(Pixel.objects.count(), 0)
+        parser.save()
+        self.assertEqual(Pixel.objects.count(), 7)
```

Comment:
we should expect 7 because? and not 14 I suppose? I think a comment or two would be nice to understand why there might be a problem here.